lua/telescope/builtin/internal.lua

```diff
@@ -231,9 +231,35 @@ internal.loclist = function(opts)
 end
 
 internal.oldfiles = function(opts)
-  local results = vim.tbl_filter(function(val)
-    return 0 ~= vim.fn.filereadable(val)
-  end, vim.v.oldfiles)
+  opts.include_current_session = utils.get_default(opts.include_current_session, true)
+
+  local results = {}
+
+  if opts.include_current_session then
+    for _, buffer in ipairs(vim.split(vim.fn.execute(':buffers! t'), "\n")) do
+      local match = tonumber(string.match(buffer, '%s*(%d+)'))
+      if match then
+        local file = vim.api.nvim_buf_get_name(match)
+        if vim.loop.fs_stat(file) and match ~= vim.api.nvim_get_current_buf() then
+          table.insert(results, file)
+        end
+      end
+    end
+  end
+
+  for _, file in ipairs(vim.v.oldfiles) do
+    if vim.loop.fs_stat(file) and vim.fn.index(results, file) == -1 then
```

Comment:
Chant we use `vim.tbl_contains(t, value)` for this. So a lua func over a viml one. Should return true if contains and false if not. (`:help tbl_contains`)

Otherwise LGTM 
Thanks :)

Reply:
Good one, pardon my VimL 😂